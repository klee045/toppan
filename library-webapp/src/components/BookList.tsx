import React, { CSSProperties } from "react";
import Book from "./Book";

const BookListStyle: CSSProperties = {
  outlineStyle: "solid",
  outlineWidth: "1px",
  width: "30%",
  display: "flex",
  flexDirection: "column",
  justifyContent: "space-around",
  alignItems: "center",
  alignSelf: "center",
  paddingTop: "20px",
  paddingBottom: "20px",
};

const BookList = ({
  books,
  toggleStates,
  handleBookClick,
}: {
  books: {
    author: string;
    name: string;
    borrower: string[];
  }[];
  toggleStates: boolean[];
  handleBookClick: (idx: number) => void;
}): React.ReactElement => {
  return (
    <div id="container" style={BookListStyle}>
      {books.length !== 0 ? (
        books.map((book, index) => (
          <div
            key={index + 1}
            id={`book-item-${index + 1}`}
            style={{
              display: "inline-flex",
              width: "80%",
              flexDirection: "column",
              alignItems: "center",
              flex: "1",
              marginBottom: "20px",
            }}
          >
            <Book
              idx={index + 1}
              bookName={book.name}
              authorName={book.author}
              borrowers={book.borrower}
              isToggled={toggleStates[index]}
              handleBookClick={handleBookClick}
            />
          </div>
        ))
      ) : (
        <div style={{ fontSize: "18px" }}>"No data found"</div>
      )}
    </div>
  );
};

export default BookList;
