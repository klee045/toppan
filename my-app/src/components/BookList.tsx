import React, { CSSProperties, useState } from "react";
import Book from "./Book";

const BookListStyle: CSSProperties = {
  outlineStyle: "solid",
  outlineWidth: "1px",
  width: "30%",
  display: "flex",
  flexDirection: "column",
  justifyContent: "space-around",
  alignItems: "center",
  alignSelf: "center",
  paddingTop: "20px",
  paddingBottom: "20px",
};

const BookList = ({
  books,
}: {
  books: {
    author: string;
    name: string;
    borrower: string[];
  }[];
}): React.ReactElement => {
  const [toggleStates, setToggleState] = useState([false, false, false]);

  const handleBookClick = (idx: number): void => {
    const temp: boolean[] = [false, false, false];
    temp[idx] = !toggleStates[idx];
    setToggleState([...temp]);
  };

  return (
    <div id="container" style={BookListStyle}>
      {books.length !== 0 ? (
        books.map((book, index) => (
          <div
            key={index + 1}
            id={`book-item-${index + 1}`}
            style={{
              display: "inline-flex",
              width: "80%",
              flexDirection: "column",
              alignItems: "center",
              flex: "1",
              marginBottom: "20px",
            }}
            onClick={() => {
              handleBookClick(index);
            }}
          >
            <Book
              idx={index + 1}
              bookName={book.name}
              authorName={book.author}
              borrowers={book.borrower}
              isToggled={toggleStates[index]}
            />
          </div>
        ))
      ) : (
        <div style={{ fontSize: "18px" }}>"No data found"</div>
      )}
    </div>
  );
};

export default BookList;
