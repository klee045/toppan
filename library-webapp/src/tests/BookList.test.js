import React from "react";
import renderer from "react-test-renderer";
import BookList from "../components/BookList";
import { shallow, configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import { screen } from "@testing-library/react";

configure({ adapter: new Adapter() });

it("renders correctly", () => {
  const tree = renderer
    .create(
      <BookList
        books={[]}
        toggleStates={[false, false, false]}
        handleBookClick={""}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('has a div with id "container"', () => {
  const tree = renderer.create(
    <BookList
      books={[]}
      toggleStates={[false, false, false]}
      handleBookClick={""}
    />
  );
  expect(
    shallow(
      <BookList
        books={[]}
        toggleStates={[false, false, false]}
        handleBookClick={""}
      />
    )
      .find("div#container")
      .exists()
  ).toBe(true);
});
