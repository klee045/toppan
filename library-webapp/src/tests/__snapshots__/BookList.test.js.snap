// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  id="container"
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "center",
      "display": "flex",
      "flexDirection": "column",
      "justifyContent": "space-around",
      "outlineStyle": "solid",
      "outlineWidth": "1px",
      "paddingBottom": "20px",
      "paddingTop": "20px",
      "width": "30%",
    }
  }
>
  <div
    style={
      Object {
        "fontSize": "18px",
      }
    }
  >
    "No data found"
  </div>
</div>
`;
